cli/cli_app.go

```diff
@@ -981,7 +982,15 @@ func runTerraformInit(originalTerragruntOptions *options.TerragruntOptions, terr
 		return err
 	}
 
-	return runTerragruntWithConfig(originalTerragruntOptions, initOptions, terragruntConfig, terraformSource != nil)
+	err = runTerragruntWithConfig(originalTerragruntOptions, initOptions, terragruntConfig, terraformSource != nil)
+
+	if err == nil {
+		moduleNeedInit := util.JoinPath(terragruntOptions.WorkingDir, MODULE_NEED_INIT)
+		if util.FileExists(moduleNeedInit) {
+			return os.Remove(moduleNeedInit)
+		}
+	}
+	return err
```

Comment:
NIT: It's more canonical golang to flip the logic to check for err. I see why you did this, but it's actually harder to understand the `err = nil` condition, as the `return err` throws you off.